### PR TITLE
chore: [ADR-0004] use RadixUI

### LINKED ISSUE #734: Radix-ui usage need an ADR
Related to #678 

cc @fmvilas

### PR DESCRIPTION
**Pull request by Amzani**
**Description**

Fixes #734
Related to #730

### CODE REVIEW

**Review by KhudaDad414 — DISMISSED** [review]
LGTM!

**Comment by Amzani** [comment]
@KhudaDad414 thanks for reviewing this, could you also review https://github.com/asyncapi/studio/pull/730 as it's linked

File: doc/adr/0004-use-radix-ui.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Date: 2023-07-17
+
+## Status
+
+Accepted
+
+
+## Context
+
+In our project, we have identified the need for a set of UI components that handle not only the logical part but also the design aspects. We have evaluated existing libraries such as Radix and Headless UI, which provide components with implicit designs.
```
Comment (fmvilas) [inline doc/adr/0004-use-radix-ui.md:12]:
```suggestion
In our project, we have identified the need for a set of UI components that only handle the logical part without imposing any design aspects. We have evaluated existing libraries such as Radix and Headless UI, which provide components without implicit designs. 
```

It's actually the opposite 😄

File: doc/adr/0004-use-radix-ui.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+## Status
+
+Accepted
+
+
+## Context
+
+In our project, we have identified the need for a set of UI components that handle not only the logical part but also the design aspects. We have evaluated existing libraries such as Radix and Headless UI, which provide components with implicit designs. 
+
+We have concluded that using a ready-made library would restrict our freedom to customize the look and feel of the components. Additionally, these libraries integrate design elements into their components, making it challenging or impossible to modify them according to our specific requirements.
```
Comment (fmvilas) [inline doc/adr/0004-use-radix-ui.md:14]:
```suggestion
We have concluded that using a ready-made library would restrict our freedom to customize the look and feel of the components. Additionally, this kind of libraries integrate design elements into their components, making it challenging or impossible to modify them according to our specific requirements.
```

**Review by fmvilas — CHANGES_REQUESTED** [review]

**Review by KhudaDad414 — APPROVED** [review]

**Review by fmvilas — APPROVED** [review]

**Pull request merged**